proto/src/determined/trial/v1/trial.proto

```diff
@@ -174,10 +176,12 @@ message TrainingMetrics {
   int32 total_records = 4;
   // The number of epochs trained on when these metrics were reported.
   float total_epochs = 5;
+  // The number of records used to compute the given metrics.
+  int32 computed_records = 6;
```

Comment:
This was from @rb-determined-ai , the idea being your training metrics @ absolute batches 100 may be an aggregate of the past 10 batches, and that number, 10, can't be capture any other way in the new API (used to be in records as "num_inputs").

Reply:
1. `computed_records` seems an unclear name to me semantically.
2. I understand this information will be good to keep. But it is not necessary because this is not used in our system at all and shouldn't be.
3. Since it is not used, why choosing records over batches or epochs? This makes "records" special. We shouldn't be opinionated about what unit users want to use since we were to expose this API to the users.

My proposal for this: we just add a field “extra” and let users/harness decide what to put in it. They can set it to be "computed_records=10" or  "computed_records=10; aggregated_method=sum". How they consume the data should be determined by users themselves. And this then decides how we design the APIs.

Reply:
What about, as it is now, computed_records is just an optional field, rather than shoving it into extra/metadata or something. I agree general training metric metadata is useful though. @rb-determined-ai thoughts (on PR for posterity?)?

Reply:
> 3. Since it is not used, why choosing records over batches or epochs? This makes "records" special. We shouldn't be opinionated about what unit users want to use since we were to expose this API to the users.

It is special because it's already in the system as `num_inputs`, so it would be a breaking change to remove it.  I mean... it's a breaking change to rename it to, but at least it's still there.  (I think there's only a few users calling `det t describe --json --metrics` to read `num_inputs` info right now).

But overall, I've heard half a dozen people complain that the naming of fields we've chosen is not clear (total_batches, computed_records, etc).  I actually think we should solve that issue first.

Then we can solve the issue of where it goes in the API.  I agree that the way it is written is not ideal.  I'm not sure I like distinguishing between metrics and metadata though; w&b just treats [metadata as other metrics](https://docs.wandb.ai/ref/python/log) and that lets them do things like support arbitrary X axes.  I don't think we can quite make that work, since we need a known x-axis to archive metrics correctly in train-crash-retrain scenarios.  But I'd be happy to be proven wrong here.

I suspect that hpviz and hparam comparisons in general also require a known x-axis, but I know less about that.